aten/src/ATen/native/quantized/cpu/qpixel_shuffle.cpp

```diff
@@ -0,0 +1,107 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/Parallel.h>
+#include <ATen/native/quantized/cpu/quantized_ops.h>
+#include <ATen/native/cpu/utils.h>
+
+#include <c10/util/irange.h>
+
+namespace at {
+namespace native {
+
+namespace {
+
+Tensor quantized_pixel_shuffle_impl(
+    const Tensor& self,
+    int64_t upscale_factor) {
+
+  TORCH_CHECK(
+      upscale_factor > 0,
+      "Number of upscale_factor to divide channels in must be positive.",
+      " Value of upscale_factor:", upscale_factor);
+  TORCH_CHECK(
+      self.dim() == 4,
+      "quantized_pixel_shuffle expects 4D input, but got input with sizes ",
+      self.sizes());
+  TORCH_CHECK(
+      self.scalar_type() == kQUInt8,
+      "quantized_pixel_shuffle works only on uint8_t.",
+      "But got:", self.scalar_type());
+
+  int64_t S = upscale_factor;
+  int64_t nbatch = self.size(0);
+  int64_t channels = self.size(1);
+  int64_t height = self.size(2);
+  int64_t width = self.size(3);
+  int64_t out_channels = channels / (S * S);
+  TORCH_CHECK(
+      channels % (S * S) == 0,
+      "quantized_pixel_shuffle expects its input's 'channel' dimension to be divisible by the square of "
+      "upscale_factor, but input.size(-3)=", channels, " is not divisible by ", (S * S));
+
+  const Tensor self_nhwc = self.contiguous(MemoryFormat::ChannelsLast);
+  Tensor qy = at::native::empty_affine_quantized(
+      {nbatch, out_channels, height * S, width * S},
+      kQUInt8,
+      c10::nullopt /* layout */,
+      kCPU,
+      c10::nullopt /* pin_memory */,
+      self_nhwc.q_scale(),
+      self_nhwc.q_zero_point(),
+      MemoryFormat::ChannelsLast);
+
+  auto qx_data = self_nhwc.data_ptr<c10::quint8>();
```

Comment:
nit: I think we can move reinterpret_cast here, same for qy_data

Reply:
yes, it should be a static cast